File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys

string = sys.stdin.readline().strip()
temp = list(string.split('-'))

result = []
for i in range(len(temp)):
    temp_str = list(temp[i].split('+'))
    sum_result = 0
    if len(temp_str) != 1:
        for j in range(len(temp_str)):
            sum_result += int(temp_str[j])
    else:
        sum_result = int(temp_str[0])
    temp[i] = sum_result

final = 0
for i in range(len(temp)):
    if i == 0:
        final += temp[i]
    else:
        final -= temp[i]
print(final)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        String[] s = bf.readLine().split("-");
        int result = 0;

        for(int i=0; i<s.length; i++){
            int temp = sum(s[i]);
            if(i==0) result += temp;
            else result -= temp;
        }

        System.out.println(result);
    }

    public static int sum(String s){
        String[] s1 = s.split("\\+");
        int sum = 0;
        for(int i=0; i<s1.length; i++){
            sum += Integer.parseInt(s1[i]);
        }

        return sum;
    }
}